src/lib/encoding.ts

```diff
@@ -329,9 +360,11 @@ export function parseManifestStream(stream: Readable): EventEmitter {
       pathCount++;
       const p = keyPath[1];
       emitter.emit('path', { path: p, id: data });
-      if (indexPath === undefined) {
-        paths[p] = data; // Maintain map of paths for use later
-      } else if (p === indexPath) {
+      if (indexProps.path === undefined) {
+        if (indexProps.id !== undefined) {
```

Comment:
Is this condition correct? Don't we want the reverse? (only keep paths if there's no index ID specified) We should also probably add unit tests for these cases.